pkg/ebpf/dns_stats.go

```diff
@@ -34,36 +38,78 @@ type dnsPacketInfo struct {
 	type_         DNSPacketType
 }
 
+type stateKey struct {
+	key dnsKey
+	id  uint16
+}
+
 type dnsStatKeeper struct {
-	mux   sync.Mutex
-	stats map[dnsKey]dnsStats
+	mux              sync.Mutex
+	stats            map[dnsKey]dnsStats
+	state            map[stateKey]time.Time
+	expirationPeriod time.Duration
+	exit             chan struct{}
 }
 
-func newDNSStatkeeper() *dnsStatKeeper {
-	return &dnsStatKeeper{
-		stats: make(map[dnsKey]dnsStats),
+func newDNSStatkeeper(timeout time.Duration) *dnsStatKeeper {
```

Comment:
In my opinion we should definitely enforce a maximum number of entries here. 
We can either drop the a new entry when cache has filled up or alternatively replace a random one.